lib/resty/dns/resolver.lua

```diff
@@ -781,10 +853,14 @@ function _M.query(self, qname, opts)
 
             if buf then
                 local answers
-                answers, err = parse_response(buf, id)
+                answers, err = parse_response(buf, id, opts)
                 if not answers then
                     if err == "truncated" then
-                        return _tcp_query(self, query, id)
+                        answers, err = _tcp_query(self, query, id, opts)
+                        if not answers then
+                          return nil, err
+                        end
+                        return answers
```

Comment:
Why eliminating the original tail call to `_tcp_query` above? Seems like we should just do

```lua
 return _tcp_query(self, query, id, opts)
```

here?